core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileSourceConfig.java

```diff
@@ -137,6 +151,21 @@ protected void validate() {
       throw new IllegalArgumentException("pathField can only be used if inputFormatClass is " +
                                            CombinePathTrackingInputFormat.class.getName());
     }
+    if (format == null || !(format.equalsIgnoreCase("text") ||
+      (format.equalsIgnoreCase("avro")) || format.equalsIgnoreCase("parquet"))) {
+      throw new IllegalArgumentException("Format can only be 'text', 'avro' or 'parquet'");
+    }
+    if (!format.equalsIgnoreCase("text")) {
+      try {
+        new Schema.Parser().parse(inputSchema);
```

Comment:
there needs to be additional validation after this. If the pathField is specified, you need to validate that it is in the schema and it is either a nullable string or a string.